server/src/test/java/org/elasticsearch/common/xcontent/XContentTypeTests.java

```diff
@@ -84,4 +84,11 @@ public void testFromRubbish() throws Exception {
         assertThat(XContentType.fromMediaTypeOrFormat("text/plain"), nullValue());
         assertThat(XContentType.fromMediaTypeOrFormat("gobbly;goop"), nullValue());
     }
+
+    public void testMediaType() throws Exception {
+        String mediaType = XContentType.parseMediaType("application/vnd.elasticsearch+json;compatible-with=7");
+        assertThat(mediaType,equalTo("application/json"));
+        mediaType = XContentType.parseMediaType("application/json");
+        assertThat(mediaType,equalTo("application/json"));
+    }
```

Comment:
can you also test yaml,cbor, and smile ?

also, does case matter ? (can you add a test for that too)